### PR TITLE
Lists updates: curate lists and blocklists

### PR DESCRIPTION
**Pull request by pfrazee**
This PR:

- Introduces non-moderation User Lists which can be used to drive feeds
- Renames Mute Lists to Mod Lists and adds the option to mute or block them
- Adds the ability to pin User Lists like feeds

Technical updates:

- Updated the list model to distinguish between curatelist and modlist
- Reworked pinned feed management to abstract over feedgens and curatelists
- Added handling of users blocked by lists
- Created a general-purpose `Tabs` component for creating tabbed interfaces with headers
- Used the tabs component to rework the ProfileList and ProfileFeed screens; consequently both have gotten a UI rework
- Added a new modal for finding and adding people to lists

Still todo:

- [x] Integration tests
- [x] New analytics events
- ~Moderation labels on lists~ blocked by backend, filed an issue

### CODE REVIEW

**Review by estrattonbailey — COMMENTED** [review]
Works great! No visible issues so far, so pumped for this. Just submitting a partial review, will keep on it tomorrow ✌️

File: src/lib/async/accumulate.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+) => Promise<AccumulateResponse<T>>
+
+export async function accumulate<T>(
+  fn: AccumulateFetchFn<T>,
+  pageLimit = 100,
+): Promise<T[]> {
+  let cursor: string | undefined
+  let acc: T[] = []
+  for (let i = 0; i < pageLimit; i++) {
+    const res = await fn(cursor)
```
[the remark this answers is not in the document]
Comment (estrattonbailey) [reply src/lib/async/accumulate.ts:17]: Nope, I see the usage 👍

File: src/Navigation.tsx [hunk 2]
```diff
@@ -168,13 +174,13 @@ function commonScreens(Stack: typeof HomeTab, unreadCountLabel?: string) {
...
-        getComponent={() => CustomFeedScreen}
+        name="ProfileFeed"
+        getComponent={() => ProfileFeedScreen}
         options={{title: title('Feed')}}
       />
       <Stack.Screen
-        name="CustomFeedLikedBy"
-        getComponent={() => CustomFeedLikedByScreen}
+        name="ProfileFeedLikedBy"
+        getComponent={() => ProfileFeedLikedByScreen}
```
[the remark this answers is not in the document]
Comment (pfrazee) [reply src/Navigation.tsx:183]: Yeah that's the idea

File: src/lib/analytics/types.ts [hunk 3]
```diff
@@ -97,10 +97,13 @@ interface TrackPropertiesMap {
   // LISTS events
   'Lists:onRefresh': {}
   'Lists:onEndReached': {}
-  'CreateMuteList:AvatarSelected': {}
-  'CreateMuteList:Save': {} // CAN BE SERVER
-  'Lists:Subscribe': {} // CAN BE SERVER
-  'Lists:Unsubscribe': {} // CAN BE SERVER
+  'CreateList:AvatarSelected': {}
+  'CreateList:SaveCurateList': {} // CAN BE SERVER
+  'CreateList:SaveModList': {} // CAN BE SERVER
```
[the remark this answers is not in the document]
Comment (pfrazee) [reply src/lib/analytics/types.ts:102]: No, we're starting over with analytics rn anyway

File: src/lib/hooks/useDesktopRightNavItems.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+interface RightNavItem {
+  uri: string
+  href: string
+  hostname: string
+  collection: string
+  rkey: string
+  displayName: string
+}
+
+export function useDesktopRightNavItems(uris: string[]): RightNavItem[] {
```
[the remark this answers is not in the document]
Comment (pfrazee) [reply src/lib/hooks/useDesktopRightNavItems.ts:16]: I guess we could; essentially we'd have to use a mobx observer and then dump it to a `setState()` to trigger the update

File: bskyweb/cmd/bskyweb/server.go
[the remark this answers is not in the document]
Comment (pfrazee) [reply bskyweb/cmd/bskyweb/server.go:1]: I'll guard against that -- it's almost certainly a missing "read after write" guarantee

[the remark this answers is not in the document]
Comment (pfrazee) [reply bskyweb/cmd/bskyweb/server.go:1]: Done, pushed

File: src/view/com/lists/ListItems.tsx [hunk 5]
```diff
@@ -124,11 +115,18 @@ export const ListItems = observer(function ListItemsImpl({
...
-        onUpdate() {
-          list.refresh()
+        onAdd(listUri: string) {
+          if (listUri === list.uri) {
+            list.cacheAddMember(profile)
+          }
+        },
+        onRemove(listUri: string) {
+          if (listUri === list.uri) {
+            list.cacheRemoveMember(profile)
```
Comment (estrattonbailey) [inline src/view/com/lists/ListItems.tsx:128]: Can these optimistic updates happen in the actual add/remove methods? Just feels odd to have them over here separate.

File: src/view/com/modals/ModerationDetails.tsx [hunk 6]
```diff
@@ -31,8 +31,25 @@ export function Component({
...
       'Moderator has chosen to set a general warning on the content.'
   } else if (moderation.cause.type === 'blocking') {
-    name = 'User Blocked'
-    description = 'You have blocked this user. You cannot view their content.'
+    if (moderation.cause.source.type === 'list') {
+      const list = moderation.cause.source.list
+      name = 'User Blocked by List'
+      description = (
+        <>
+          This user is included the{' '}
```
Comment (estrattonbailey) [inline src/view/com/modals/ModerationDetails.tsx:39]:
```suggestion
          This user is included in the{' '}
```

File: src/view/com/pager/PagerWithHeader.tsx
Comment (estrattonbailey) [inline src/view/com/pager/PagerWithHeader.tsx:1]: Seems to work great 👍

File: src/view/com/profile/ProfileSubpageHeader.tsx [hunk 7]
```diff
@@ -0,0 +1,194 @@
...
+    }
+  }, [navigation])
+
+  const onPressMenu = React.useCallback(() => {
+    store.shell.openDrawer()
+  }, [store])
+
+  const onPressAvi = React.useCallback(() => {
+    if (
+      avatar // TODO && !(view.moderation.avatar.blur && view.moderation.avatar.noOverride)
```
Comment (estrattonbailey) [inline src/view/com/profile/ProfileSubpageHeader.tsx:64]: TODO?

**Review by estrattonbailey — APPROVED** [review]
`*slow chant of paul paul paul paul PAULPAULPAUL*`

File: src/view/com/profile/ProfileSubpageHeader.tsx (hunk 7 above)
Comment (pfrazee) [reply src/view/com/profile/ProfileSubpageHeader.tsx:64]: yeah we'll come back to it when mod labels get added to lists or feeds

File: src/view/com/lists/ListItems.tsx (hunk 5 above)
Comment (pfrazee) [reply src/view/com/lists/ListItems.tsx:128]: That modal uses a different set of ListModels that aren't shared by the one showing the user, unfortunately

**Pull request merged**